contract_sale_generation/models/abstract_contract.py

```diff
@@ -0,0 +1,17 @@
+# Copyright 2017 Pesol (<http://pesol.es>)
+# Copyright 2017 Angel Moya <angel.moya@pesol.es>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import fields, models
+
+
+class ContractAbstractContract(models.AbstractModel):
+    _inherit = "contract.abstract.contract"
+
+    type = fields.Selection(
```

Comment:
@Freni-OSI IMHO, the field should be on base contract module with just 'invoice' value